recipes/mp-units/2.2.0/conanfile.py

```diff
@@ -0,0 +1,266 @@
+
+    def _is_feature_enabled(self, name):
+        compiler = self.settings.compiler
+        opt = self.options.get_safe(name)
+        feature_name = self._option_feature_map[name]
+        feature = self._feature_compatibility[feature_name]
+        min_version = feature["compiler"].get(str(compiler))
+        return bool(
+            opt == True
+            or (
+                opt == "auto"
+                and min_version
+                and not loose_lt_semver(str(compiler.version), min_version)
+            )
+        )
+
+    @property
+    def _build_all(self):
+        return bool(self.conf.get("user.mp-units.build:all", default=False))
+
+    @property
+    def _build_cxx_modules(self):
+        return self._is_feature_enabled("cxx_modules")
+
+    @property
+    def _use_fmtlib(self):
+        return not self._is_feature_enabled("std_format")
+
+    def requirements(self):
+        if not self.options.freestanding:
+            if self.options.contracts == "gsl-lite":
+                self.requires("gsl-lite/0.41.0")
+            elif self.options.contracts == "ms-gsl":
+                self.requires("ms-gsl/4.0.0")
+            if self._use_fmtlib:
+                self.requires("fmt/10.2.1")
+
+    def validate(self):
+        self._check_feature_supported("mp-units", "minimum_support")
+        for key, value in self._option_feature_map.items():
+            if self.options.get_safe(key) == True:
+                self._check_feature_supported(key, value)
+        if self.options.freestanding and self.options.contracts != "none":
+            raise ConanInvalidConfiguration(
+                "'contracts' should be set to 'none' for a freestanding build"
+            )
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self._build_cxx_modules:
+            tc.cache_variables["CMAKE_CXX_SCAN_FOR_MODULES"] = True
+            tc.cache_variables["MP_UNITS_BUILD_CXX_MODULES"] = str(
+                self.options.cxx_modules
+            ).upper()
+        if self.options.freestanding:
+            tc.cache_variables["MP_UNITS_API_FREESTANDING"] = True
+        else:
+            tc.cache_variables["MP_UNITS_API_STD_FORMAT"] = str(
+                self.options.std_format
+            ).upper()
+        tc.cache_variables["MP_UNITS_API_STRING_VIEW_RET"] = str(
+            self.options.string_view_ret
+        ).upper()
+        tc.cache_variables["MP_UNITS_API_NO_CRTP"] = str(self.options.no_crtp).upper()
+        tc.cache_variables["MP_UNITS_API_CONTRACTS"] = str(
+            self.options.contracts
+        ).upper()
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(build_script_folder="src")
+        cmake.build()
```

Comment:
If this is a header-only library, what's being built in this case? Or is it only used for the following install call=

Reply:
Right, but we now also have C++20 modules support :wink: which has to be built. However, a branch would be a better solution here.

Reply:
Ohh I see! This is something @jcar87 has been looking at, excited to finally see modules making their way in :)